src/ui/ResponsiveComponents/ResponsiveComponentsManager.ts

```diff
@@ -62,20 +62,7 @@ export class ResponsiveComponentsManager {
     const initEventRoot = options.initializationEventRoot || root;
     initEventRoot.on(InitializationEvents.afterInitialization, () => {
       if (this.shouldEnableResponsiveMode(root)) {
-        let responsiveComponentsManager = _.find(this.componentManagers, componentManager => root.el == componentManager.coveoRoot.el);
-        if (!responsiveComponentsManager) {
-          responsiveComponentsManager = new ResponsiveComponentsManager(root);
-        }
-
-        if (!Utils.isNullOrUndefined(options.enableResponsiveMode) && !options.enableResponsiveMode) {
-          responsiveComponentsManager.disableComponent(ID);
-          return;
```

Comment:
This early return prevented decrementing the counter on line 68, which prevented entering the block on line 69 needed to make the facets responsive.

Extracting the logic into a separate function creates a new stack entry. Returning early there does not prevent the rest of the logic in the parent function from running.